### PR TITLE
Organisation settings profile

### PR DESCRIPTION
**Pull request by onkar-saeloun**
What- 
PR includes 
- The design and API integration of the organization settings profile page.
- The Drag and drop image feature
- API handling and Error state management.

Why-
The new APIs and design need to be integrated as the new signup flow is completed.

Notion-
https://www.notion.so/saeloun/Frontend-Update-Organization-settings-after-API-changes-for-address-fields-and-business-phone-fiel-ed325a369f084ebbbe61c84d8973d125?pvs=4

Loom-
https://www.loom.com/share/96015c765ede4c898aec49f791522af5

### CODE REVIEW

File: app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+              onLogoChange={onLogoChange}
+            />
+          )}
+          <div className="ml-5 w-36 whitespace-pre-wrap text-left text-xs	 text-miru-dark-purple-400 ">
+            <FileAcceptanceText />
+          </div>
+        </div>
+        <div className="flex w-1/2 flex-col px-2 pt-1">
+          <CustomInputText
+            dataCy="company-name"
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx:89]: not required any more as we are not using cypress any more

Comment (prasanthchaduvula) [reply app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx:89]: Remove `dataCy` key from all the inputs and buttons from this PR file changes

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

File: app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx (hunk 1 above)
Comment (onkar-saeloun) [reply app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx:89]: Okay removing it

File: app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx [hunk 2]
```diff
@@ -0,0 +1,409 @@
+import React from "react";
+
+import {
+  CalendarIcon,
+  InfoIcon,
+  MapPinIcon,
+  MoneyIcon,
+  PhoneIcon,
+} from "miruIcons";
+import PhoneInput from "react-phone-number-input";
```
[the remark this answers is not in the document]
Comment (onkar-saeloun) [reply app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx:10]: Done

File: app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx (hunk 1 above)
Comment (onkar-saeloun) [reply app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx:89]: Done

File: app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (onkar-saeloun) [reply app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx:10]: @prasanthchaduvula getting formatting issue if declared below the react-phone-input

File: app/javascript/src/components/Profile/Layout.tsx [hunk 3]
```diff
@@ -35,19 +39,26 @@ const Layout = ({ isAdminUser, user, company }) => {
...
+      {isDesktop && (
+        <Fragment>
+          <div className="mt-3">
+            <Header />
+          </div>
+          <div className="mt-6 mb-10 grid grid-cols-12 gap-11">
+            <div className="col-span-3">
+              <SideNav
+                company={company}
+                firstName={settingsStates.profileSettings?.firstName}
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Profile/Layout.tsx:51]: Please destructure these values

File: app/javascript/src/components/Profile/Layout.tsx [hunk 4]
```diff
@@ -35,19 +39,26 @@ const Layout = ({ isAdminUser, user, company }) => {
...
+                isAdmin={isAdminUser}
+                lastName={settingsStates.profileSettings?.lastName}
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Profile/Layout.tsx:53]: ditto
Please destructure these values

File: app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx [hunk 5]
```diff
@@ -0,0 +1,409 @@
...
+      </div>
+      <div className="w-72">
+        <div className="flex w-full flex-col px-2 pb-3">
+          <CustomInputText
+            dataCy="address-line-1"
+            id="addressLine1"
+            label="Address line 1"
+            name="addressLine1"
+            type="text"
+            value={orgDetails.companyAddr.addressLine1}
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx:180]: can we destructure these values and store them

File: app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx [hunk 6]
```diff
@@ -0,0 +1,409 @@
...
+          )}
+        </div>
+        <div className="flex w-full flex-col px-2 py-3">
+          <CustomInputText
+            dataCy="address-line-2"
+            id="addressLine2"
+            label="Address line 2 (optional)"
+            name="addressLine2"
+            type="text"
+            value={orgDetails.companyAddr.addressLine2}
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx:197]: ditto
can we destructure these values and store them

File: app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx [hunk 7]
```diff
@@ -0,0 +1,409 @@
...
+          />
+        </div>
+        <div className="flex flex-row pt-2 pb-1">
+          <div className="flex w-1/2 flex-col px-2 pb-3">
+            <CustomReactSelect
+              handleOnChange={value => handleOnChangeCountry(value)}
+              label="Country"
+              name="countrySelect"
+              options={countries}
+              value={orgDetails.companyAddr.country}
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx:208]: can we destructure these values and store them

File: app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx [hunk 8]
```diff
@@ -0,0 +1,409 @@
...
+          <div className="flex w-1/2 flex-col px-2 pb-3">
+            <CustomReactSelect
+              handleOnChange={state => handleOnChangeState(state)}
+              label="State"
+              name="stateSelect"
+              options={stateList}
+              value={
+                orgDetails.companyAddr.state
+                  ? {
+                      label: orgDetails.companyAddr.state.label,
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx:226]: ditto
can we destructure these values and store them
Please make this change across file

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

File: app/javascript/src/components/Profile/Layout.tsx (hunk 3 above)
Comment (onkar-saeloun) [reply app/javascript/src/components/Profile/Layout.tsx]: done

File: app/javascript/src/components/Profile/Layout.tsx (hunk 4 above)
Comment (onkar-saeloun) [reply app/javascript/src/components/Profile/Layout.tsx]: Done

File: app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx (hunk 5 above)
Comment (onkar-saeloun) [reply app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx]: Done

File: app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx (hunk 6 above)
Comment (onkar-saeloun) [reply app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx]: Done

File: app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx (hunk 7 above)
Comment (onkar-saeloun) [reply app/javascript/src/components/Profile/Organization/Edit/StaticPage.tsx]: Done

**Review by prasanthchaduvula — APPROVED** [review]

**Pull request merged**